examples/shopping_ads/add_smart_shopping_ad.rb

```diff
@@ -0,0 +1,218 @@
+  campaign.maximize_conversion_value.target_roas = client.wrapper.double(3.5)
+  campaign.tracking_url_template = client.wrapper.string("http://example.com/track")
+
+  op = client.operation(:Campaign)
+  op["create"] = campaign
+
+  campaign_service = client.service(:Campaign)
+  response = campaign_service.mutate_campaigns(customer_id, [op])
+
+  puts("Created campaign with id #{response.results.first.resource_name}")
+
+  response.results.first.resource_name
+end
+
+def add_smart_shopping_ad_group(client, customer_id, campaign_resource_name)
+  ad_group = client.resource(:AdGroup)
+  ad_group.name = client.wrapper.string("Earth to Mars Cruises #{(Time.now.to_f*1000).to_i}")
+  ad_group.campaign = client.wrapper.string(campaign_resource_name)
+  ad_group.type = :SHOPPING_SMART_ADS
+  ad_group.status = :ENABLED
+
+  op = client.operation(:AdGroup)
+  op["create"] = ad_group
+
+  ad_group_service = client.service(:AdGroup)
+  response = ad_group_service.mutate_ad_groups(customer_id, [op])
+
+  puts("Created smart shopping ad group with id #{response.results.first.resource_name}")
+  response.results.first.resource_name
+end
+
+def add_shopping_listing_group(client, customer_id, ad_group_resource_name)
+  ad = client.resource(:Ad)
+  ad.shopping_smart_ad = client.resource(:ShoppingSmartAdInfo)
+
+  ad_group_ad = client.resource(:AdGroupAd)
+  ad_group_ad.ad = ad
+  ad_group_ad.ad_group = client.wrapper.string(ad_group_resource_name)
+
+  op = client.operation(:AdGroupAd)
+  op["create"] = ad_group_ad
+
+  ad_group_ad_service = client.service(:AdGroupAd)
+  response = ad_group_ad_service.mutate_ad_group_ads(customer_id, [op])
+
+  puts("Created ad group ad with id #{response.results.first.resource_name}")
+end
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
+  options[:merchant_center_account_id] = 'INSERT_MERCHANT_CENTER_ACCOUNT_ID_HERE'
+  options[:create_default_listing_group] = false
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf("Usage: #{__FILE__} [options]")
+
+    opts.separator ''
+    opts.separator 'Options:'
+
+    opts.on('-C', '--customer-id CUSTOMER-ID', String, 'Customer ID') do |v|
+      options[:customer_id] = v
+    end
+
+    opts.on(
+      '-M',
+      '--customer-id MERCHANT-CENTER-ACCOUNT-ID',
+      String,
+      'Merchant center account id'
+    ) do |v|
+      options[:merchant_center_account_id] = v
+    end
+
+
+    opts.on('-l', '--create-default-listing-group CREATE-DEFAULT-LISTING-GROUP', String, 'create default listing group') do |v|
+      options[:create_default_listing_group] = if v == "true"
```

Comment:
WDYT of the convention established here for boolean command line flags: https://github.com/googleads/google-ads-ruby/blob/master/examples/shopping_ads/add_shopping_product_ad.rb#L215

Reply:
that'll create a truthy value if the user passes the literal string "false", the canonical correct way to do this in ruby these days is:

https://github.com/rails/rails/blob/master/activemodel/lib/active_model/type/boolean.rb

but I'm not sure we want to pull in activemodel or vendor this code in just for code samples. WDYT?

Reply:
We definitely don't want to include any new dependencies for this. Can you update other examples that require true/false values to use whatever paradigm we go with here?

You can simplify this to: `options[:...] = v == "true"`